cmd/limactl/snapshot.go

```diff
@@ -0,0 +1,161 @@
+)
+
+func newSnapshotCommand() *cobra.Command {
+	var snapshotCmd = &cobra.Command{
+		Use:   "snapshot",
+		Short: "Manage instance snapshots",
+	}
+	snapshotCmd.AddCommand(newSnapshotApplyCommand())
+	snapshotCmd.AddCommand(newSnapshotCreateCommand())
+	snapshotCmd.AddCommand(newSnapshotDeleteCommand())
+	snapshotCmd.AddCommand(newSnapshotListCommand())
+
+	return snapshotCmd
+}
+
+func newSnapshotCreateCommand() *cobra.Command {
+	var createCmd = &cobra.Command{
+		Use:               "create INSTANCE",
+		Aliases:           []string{"save"},
+		Short:             "Create (save) a snapshot",
+		Args:              cobra.MinimumNArgs(1),
+		RunE:              snapshotCreateAction,
+		ValidArgsFunction: snapshotBashComplete,
+	}
+	createCmd.Flags().String("tag", "", "name of the snapshot")
+
+	return createCmd
+}
+
+func snapshotCreateAction(cmd *cobra.Command, args []string) error {
+	instName := args[0]
+
+	inst, err := store.Inspect(instName)
+	if err != nil {
+		return err
+	}
+
+	tag, err := cmd.Flags().GetString("tag")
+	if err != nil {
+		return err
+	}
+
+	if tag == "" {
+		return fmt.Errorf("expected tag")
+	}
+
+	ctx := cmd.Context()
+	return snapshot.Save(ctx, inst, tag)
+}
+
+func newSnapshotDeleteCommand() *cobra.Command {
+	var deleteCmd = &cobra.Command{
+		Use:               "delete INSTANCE",
+		Aliases:           []string{"del"},
+		Short:             "Delete (del) a snapshot",
+		Args:              cobra.MinimumNArgs(1),
+		RunE:              snapshotDeleteAction,
+		ValidArgsFunction: snapshotBashComplete,
+	}
+	deleteCmd.Flags().String("tag", "", "name of the snapshot")
+
+	return deleteCmd
+}
+
+func snapshotDeleteAction(cmd *cobra.Command, args []string) error {
+	instName := args[0]
+
+	inst, err := store.Inspect(instName)
+	if err != nil {
+		return err
+	}
+
+	tag, err := cmd.Flags().GetString("tag")
+	if err != nil {
+		return err
+	}
+
+	if tag == "" {
+		return fmt.Errorf("expected tag")
+	}
+
+	ctx := cmd.Context()
+	return snapshot.Del(ctx, inst, tag)
+}
+
+func newSnapshotApplyCommand() *cobra.Command {
+	var applyCmd = &cobra.Command{
+		Use:               "apply INSTANCE",
+		Aliases:           []string{"load"},
+		Short:             "Apply (load) a snapshot",
+		Args:              cobra.MinimumNArgs(1),
+		RunE:              snapshotApplyAction,
+		ValidArgsFunction: snapshotBashComplete,
+	}
+	applyCmd.Flags().String("tag", "", "name of the snapshot")
+
+	return applyCmd
+}
+
+func snapshotApplyAction(cmd *cobra.Command, args []string) error {
+	instName := args[0]
+
+	inst, err := store.Inspect(instName)
+	if err != nil {
+		return err
+	}
+
+	tag, err := cmd.Flags().GetString("tag")
+	if err != nil {
+		return err
+	}
+
+	if tag == "" {
+		return fmt.Errorf("expected tag")
+	}
+
+	ctx := cmd.Context()
+	return snapshot.Load(ctx, inst, tag)
+}
+
+func newSnapshotListCommand() *cobra.Command {
+	var listCmd = &cobra.Command{
+		Use:               "list INSTANCE",
+		Short:             "List existing snapshots",
```

Comment:
Perhaps add `Long` message to clarify that the result is not machine-parsable

Reply:
Theoretically we _could_ parse it, it is a space separated list with a column header.

https://github.com/qemu/qemu/blob/v7.0.0/block/qapi.c#L664_L698

Reply:
I added a simple --quiet command, to only print the tags (but didn't bother with --json)